pkg/util/indexparamcheck/hnsw_checker.go

```diff
@@ -31,7 +31,7 @@ func (c hnswChecker) CheckTrain(params map[string]string) error {
 }
 
 func (c hnswChecker) CheckValidDataType(dType schemapb.DataType) error {
-	if dType != schemapb.DataType_FloatVector && dType != schemapb.DataType_BinaryVector && dType != schemapb.DataType_Float16Vector {
+	if dType != schemapb.DataType_FloatVector && dType != schemapb.DataType_BinaryVector && dType != schemapb.DataType_Float16Vector && dType != schemapb.DataType_BFloat16Vector {
```

Comment:
!IsVectorType(dType)

Reply:
In fact, the IsVectorType() function is defined in "internal/proxy/utils.go". Maybe I should find an appropriate location in "pkg/util" and define it again.